fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/fragments/ManageElectionFragment.java

```diff
@@ -75,43 +113,229 @@ public View onCreateView(
     @Override
     public void onActivityCreated(@Nullable Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
-        setupHomeButton();
         Button back = (Button) getActivity().findViewById(R.id.tab_back);
-        back.setOnClickListener(c->laoDetailViewModel.openLaoDetail());
-        //On click, terminate button  current Election
-        terminate.setOnClickListener(
+        back.setOnClickListener(v -> laoDetailViewModel.openLaoDetail());
+
+
+        DialogInterface.OnClickListener dialogClickListener = (dialog, which) -> {
+            switch (which) {
+                case DialogInterface.BUTTON_POSITIVE: {
+                    //Yes button clicked
+                    switch (requestCode) {
+                        case CANCEL_CODE: {
+                            // TODO : In the future send a UpdateElection message with a modified end time as the current time
+                            laoDetailViewModel.openLaoDetail();
+                            break;
+                        }
+
+
+                        case START_TIME_CODE: {
+                            Calendar startTimeCalendar = Calendar.getInstance();
+                            startTimeCalendar.setTimeInMillis(laoDetailViewModel.getCurrentElection().getStartTimestamp());
```

Comment:
Why milliseconds and not seconds?

Reply:
because there is only a method setTimeInMillis. To fix this I did *1000 to the timestamp .